samples/WebApi/README.md

```diff
@@ -0,0 +1,136 @@
+
+```xml
+<add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" /> 
+<add key="aspnet-core" value="https://dotnet.myget.org/F/aspnetcore-release/api/v3/index.json" /> 
+```
+
+Once you've added the package source, add a reference to the compiler by running the following command:
+
+`> dotnet add package Microsoft.DotNet.ILCompiler -v 1.0.0-alpha-* `
+
+## Add Core MVC services
+With the package successfully added to your project, your project's default registered MVC services must be modified.
+
+The default template's `AddMvc()` call registers a large set of middleware services by default, even if they are not needed by your application. This is not AOT-compilation friendly because it leads to large binaries and creates the risk of adding unsupported features.
+
+Open the file called `Startup.cs` and in the `ConfigureServices()` method and modify the line:
+
+```csharp
+services.AddMvc();
+```
+
+to
+
+```csharp
+services.AddMvcCore().AddJsonFormatters();
+```
+
+Replacing `AddMvc()` with `AddMvcCore()` adds only the basic MVC functionality. It's followed by explicit registration of services, which are required by the application - in this case the `JsonFormatter`. For more details see [Fabian Gosebrink's blog post comparing AddMvc and AddMvcCore](https://dzone.com/articles/the-difference-between-addmvc-and-addmvccore).
+
+## Using reflection 
+Runtime directives are XML configuration files, which specify which elements of your program are available for reflection. They are used at compile-time to enable AOT compilation in applications at runtime. 
+
+In this sample a basic rd.xml file has been added for a simple Web API application under the root project folder. Copy its contents to your application directory and modify the
+```xml <Assembly Name="SampleWebApi" Dynamic="Required All" /> ``` element to use your app's name.
+
+If your application makes use of reflection, you will need to create a rd.xml file specifying explicitly which assemblies and types should be made available. For example, in  your .NET Core Web API application, reflection is required to determine the correct namespace, from which to load the ``Startup`` type. Both are defined respectively via the `<Assembly>` and `<Type>` attributes. For example, in the case of our specific application;
+
+```xml 
+<Assembly Name="SampleWebApi">
+  <Type Name="SampleWebApi.Startup" Dynamic="Required All" />
+</Assembly>
+```
+
+At runtime, if a method or type is not found or cannot be loaded, an exception will be thrown. The exception message will contain information on the missing type reference, which you can then add to the rd.xml of your program.
+
+Once you've created a rd.xml file, navigate to the root directory of your project and open its `.csproj` file and in the first `<PropertyGroup>` element add the following:
+
+```xml
+<RdXmlFile>path_to_rdxml_file\rd.xml</RdXmlFile>
+<RuntimeIdentifiers>runtime_identifier</RuntimeIdentifiers>
```

Comment:
Maybe the RuntimeIdentifier part should be in a separate section? It's unrelated to using reflection.

Reply:
Separated into two sentences, but in the same section (so following the guide doesn't involve opening and closing the same files).